skbio/io/format/embl.py

```diff
@@ -0,0 +1,1568 @@
+                else:
+                    reference += wrapper.wrap(record)
+
+        # add a spacer between references (but no at the final reference)
+        # cause the caller will add spacer
+        if (i+1) < len(obj):
+            reference += ["XX"]
+
+    # now define a string and add a final "\n"
+    s = "\n".join(reference) + "\n"
+
+    # and return it
+    return s
+
+
+# parse an embl reference record.
+def _parse_source(lines):
+    '''Parse single SOURCE field.
+    '''
+
+    # parsed reference will be placed here
+    res = {}
+
+    # define a section splitter with _embl_yield_section function defined in
+    # this module
+    section_splitter = _embl_yield_section(lambda line: _get_embl_key(line),
+                                           skip_blanks=True, strip=False)
+
+    # now itereta along sections (lines of the same type)
+    for section in section_splitter(lines):
+        # this function append all data in the same keywords. A list of lines
+        # as input (see skbio.io.format._sequence_feature_vocabulary)
+        label, data = _embl_parse_section_default(
+            section, join_delimiter=' ', return_label=True)
+
+        res[label] = data
+
+    # return translated keys
+    return _translate_keys(res)
+
+
+def _serialize_source(header, obj, indent=5):
+    '''Serialize SOURCE.
+
+    Parameters
+    ----------
+    header: section header
+    obj : dict
+    indent : indent length
+    '''
+
+    source = []
+
+    # treat taxonomy and all others keys
+    for key in ["ORGANISM", "taxonomy", "organelle"]:
+        # get data to serielize
+        data = obj.get(key)
+
+        # if key is not defined (eg. organelle, continue)
+        if data is None:
+            continue
+
+        # get embl key for my key (eg, taxonomy -> OC)
+        embl_key = REV_KEYS_TRANSLATOR.get(key, key)
+
+        # get an embl wrapper
+        wrapper = _get_embl_wrapper(embl_key, indent)
+
+        # define wrapped string
+        source += wrapper.wrap(data)
+
+    # now define a string and add a final "\n"
+    s = "\n".join(source) + "\n"
+
+    # and return it
+    return s
+
+
+def _parse_sequence(lines):
+    '''Parse the sequence section for sequence.'''
+
+    # when reading a feature-level-products accession, features are relative
+    # to parent accession, so we need to parse accessions like
+    # LK021130.1:74067..75610:rRNA to model a table feature like
+    # FT   rRNA            LK021130.1:74067..75610
+    # FT                   /gene="16S"
+    # FT                   /product="16S rRNA"
+    # FT                   /note="16S rRNA subunit (checked and believed
+    # FT                   to be right,based on 454- and PacBio-sequencing)"
+    # I could express features by resizing sequence, however the resulting
+    # sequence will be different than data read. I decide to mantain
+    # sequences indentical to read values, and to discard features
+    # for such entries
```

Comment:
Removed. This comment and the previous one were added when I dealt with feature-level-products,  I was thinking to resize sequence to define interval metadata for such records. Then I chose to ignore completely interval metadata.